src/query/scan_manager.c

```diff
@@ -8117,7 +8225,7 @@ check_hash_list_scan (LLIST_SCAN_ID * llsidp, int *val_cnt, int hash_list_scan_y
   UINT64 mem_limit = prm_get_bigint_value (PRM_ID_MAX_HASH_LIST_SCAN_SIZE);
 
   /* no_hash_list_scan sql hint check */
-  if (hash_list_scan_yn == 0)
+  if (hash_list_scan_yn == HASH_METH_NOT_USE)
```

Comment:
The function signature should be like this?
from
```cpp
check_hash_list_scan (..., int hash_list_scan_yn)
```
to
```cpp
check_hash_list_scan (..., HASH_METHOD hash_list_scan_yn)
```

Reply:
Because of 'XASL to STREAM', it is better to keep hash_list_scan_yn as int.
In executor, the hash_list_scan_type is 'HASH_METHOD'.

Reply:
Then, how about adding assert()? to check the `hash_list_scan_yn` value is out of HASH_METHOD values possibly.

Reply:
OK. I'll add assert().